### PR TITLE
8329141: Obsolete RTM flags and code

### PR DESCRIPTION
**Pull request by vnkozlov**
Obsolete HotSpot RTM flags which were deprecated in JDK 23.
RTM related VM code and tests were removed.

Tested tier1-3,stress,xcomp

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8333763](https://bugs.openjdk.org/browse/JDK-8333763) to be approved

### Issues
 * [JDK-8329141](https://bugs.openjdk.org/browse/JDK-8329141): Obsolete RTM flags and code (**Enhancement** - P4)
 * [JDK-8333763](https://bugs.openjdk.org/browse/JDK-8333763): Obsolete RTM flags and code (**CSR**)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [fcfb62fd](https://git.openjdk.org/jdk/pull/19589/files/fcfb62fd57e71b29a7360993e577a25e126ad044)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19589/head:pull/19589` \
`$ git checkout pull/19589`

Update a local copy of the PR: \
`$ git checkout pull/19589` \
`$ git pull https://git.openjdk.org/jdk.git pull/19589/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19589`

View PR using the GUI difftool: \
`$ git pr show -t 19589`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19589.diff">https://git.openjdk.org/jdk/pull/19589.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19589#issuecomment-2153724533)

### CODE REVIEW

**Review by chhagedorn — APPROVED** [review]
Otherwise, looks good to me.

File: src/hotspot/share/adlc/output_c.cpp [hunk 1]
```diff
@@ -1614,11 +1614,6 @@ void ArchDesc::defineExpand(FILE *fp, InstructForm *node) {
         fprintf(fp, "  ((MachIfNode*)n%d)->_fcnt = _fcnt;\n", cnt);
       }
 
-      if (node->is_ideal_fastlock() && new_inst->is_ideal_fastlock()) {
```
[the remark this answers is not in the document]
Comment (vnkozlov) [reply src/hotspot/share/adlc/output_c.cpp:1617]: Done

File: src/hotspot/share/opto/parse1.cpp [hunk 2]
```diff
@@ -2201,42 +2198,6 @@ void Parse::clinit_deopt() {
...
-    Node* adr_node = basic_plus_adr(mdo, mdo, offset);
-    Node* rtm_state = make_load(control(), adr_node, TypeInt::INT, T_INT, adr_type, MemNode::unordered);
-
-    // Separate Load from Cmp by Opaque.
-    // In expand_macro_nodes() it will be replaced either
-    // with this load when there are locks in the code
-    // or with ProfileRTM (cmp->in(2)) otherwise so that
-    // the check will fold.
-    Node* profile_state = makecon(TypeInt::make(ProfileRTM));
-    Node* opq   = _gvn.transform( new Opaque3Node(C, rtm_state, Opaque3Node::RTM_OPT) );
```
[the remark this answers is not in the document]
Comment (vnkozlov) [reply src/hotspot/share/opto/parse1.cpp:2228]: Done

**Pull request closed without merging**